### PR TITLE
SF6: Use AbstractAuthenticator for TokenAuthenticator

### LINKED ISSUE #34762: Migrate TokenAuthenticator to the new SF system
- [ ] TokenAuthenticator is based on an old guard system `AbstractGuardAuthenticator`, check if it is already done to rely on the new system (if possible, since we may not rely on it yet) ortherwise migrate to the new system
Requirement for API V1

### PR DESCRIPTION
**Pull request by M0rgan01**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Update anonymous access, see https://symfony.com/doc/current/security.html#allowing-unsecured-access-i-e-anonymous-users. Use AbstractAuthenticator instead of AbstractGuardAuthenticator (deprecated)
| Type?             | improvement
| Category?         | BO
| BC breaks?        | yes
| Deprecations?     | no
| How to test?      | CI and UI tests are green
| UI Tests          | https://github.com/M0rgan01/ga.tests.ui.pr/actions/runs/7197890031
| Fixed issue or discussion?     | Fixes #34762
| Related PRs       | -
| Sponsor company   | -

### CODE REVIEW

File: src/Core/Security/OAuth2/TokenAuthenticator.php [hunk 1]
```diff
@@ -71,58 +61,54 @@ public function start(Request $request, AuthenticationException $authException =
...
-            }
-            $explode = explode(' ', $authorization);
-            if (count($explode) >= 2) {
-                $token = $explode[1];
+        $authorization = $request->headers->get('Authorization') ?? null;
+        if (null === $authorization) {
+            return false;
+        }
+        $explode = explode(' ', $authorization);
+        if (count($explode) >= 2) {
```
Comment (jolelievre) [inline src/Core/Security/OAuth2/TokenAuthenticator.php]: It was already wrong initially, but there should be an `else` statement that returns false if there are not two parts

**Review by matthieu-rolland — APPROVED** [review]
lgtm 👍

**Review by jolelievre — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]
:ok: for UI Tests (Ping @PrestaShop/ui-tests-maintainers)

**Comment by jolelievre** [comment]
Thanks @M0rgan01

**Pull request merged**